inspirehep/dojson/utils.py

```diff
@@ -158,7 +158,10 @@ def get_record_ref(recid, record_type='record'):
         server = current_app.config.get('SERVER_NAME', default_server)
     else:
         server = default_server
-    return {'$ref': 'http://{}/api/{}/{}'.format(server, record_type, recid)}
+    # This config might also be http://inspirehep.net.
+    if not server.startswith('http://'):
```

Comment:
SERVER_NAME could be indeed either http or https depending on the specific installation.